src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -359,15 +370,45 @@ public void Train(double[] values, ref double[][] results)
                 {
                     Array.Resize<double[]>(ref results, values.Length);
                 }
-                SpectralResidual(values, results, _threshold);
+
+                bool isTemporal = true;
+                double[] seriesToDetect = values.ToArray();
```

Comment:
>ToArray [](start = 49, length = 7)

`values` is already an array. Is the intention to copy it to a new array? If yes, please use the `AllocateDoubleArray` method and define a field for `seriesToDetect`, similar to how other arrays are allocated in this class. This is to prevent additional allocations for every computation, which would cause the GC to work overtime.

Reply:
updated.

---
In reply to: [435218315](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435218315) [](ancestors = 435218315)